web/react/components/rename_channel_modal.jsx

```diff
@@ -95,6 +95,9 @@ module.exports = React.createClass({
             var button = $(e.relatedTarget);
             self.setState({ display_name: button.attr('data-display'), title: button.attr('data-name'), channel_id: button.attr('data-channelid') });
         });
+        $(this.refs.modal.getDOMNode()).on('hidden.bs.modal', function() {
+            self.setState({display_name: "", channel_name: "", display_name_error: "", server_error: "", name_error: ""});
+        });
```

Comment:
Yeah, I think it's good practice to have them, in case we decide to do something wild in the future like loading new pages in place, etc.